pii_data.py

```diff
@@ -58,16 +58,29 @@ def has_ipv4(self, anonymize=False):
             return ipv4
         return bool(count2 + count1)
 
-    def has_ipv6(self):
-        match = re.search(r'(^(\b[0-9a-fA-F]{0,4}\b)?:(\b[0-9a-fA-F]{0,4}\b)?:'
-                          r'(\b[0-9a-fA-F]{0,4}\b)?:(\b[0-9a-fA-F]{0,4}\b)?:'
-                          r'(\b[0-9a-fA-F]{0,4}\b)?:(\b[0-9a-fA-F]{0,4}\b)?:'
-                          r'(\b[0-9a-fA-F]{0,4}\b)?:(\b[0-9a-fA-F]{0,4}\b)?$)', self)
+    def has_ipv6(self, anonymize=False):
+
+        ipv6, count = re.subn(r'((\w((?:[0-9a-fA-F]?){0,4}:)|(:)){7}((?:[0-9a-fA-F]?){0,4}))([^:[0-9a-fA-F]])*',
+                              '[iPv6 address]', self)
+
+        ipv6, count0 = re.subn(r'((\w((?:[0-9a-fA-F]?){0,4}:)|(:)){7}((?:[0-9a-fA-F]?){0,4}))([^:[0-9a-fA-F]])*',
+                               '[iPv6 address]', ipv6)
         if self.__eq__('0:0:0:0:0:0:0:0') | self.__eq__(':::::::'):
```

Comment:
You've used a [bitwise](https://www.w3schools.com/python/gloss_python_bitwise_operators.asp) operator here - you don't want to do that.  Instead you should use or like this:
```
        if self.__eq__('0:0:0:0:0:0:0:0') or self.__eq__(':::::::'):
```